core/domain/classifier_services.py

```diff
@@ -127,6 +127,52 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        create_classifier_training_job(algorithm_id, interaction_id,
+                                       exp_id, exp_version, state_name,
+                                       training_data,
+                                       feconf.TRAINING_JOB_STATUS_NEW)
+
+
+def update_classifier_exploration_mappings(exploration, state_names):
+    """Creates new ClassifierExplorationMappingModel instances for all the
+    state names passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
```

Comment:
So, if I'm reading this right, you are retrieving the classifier using get_classifier() which uses the existing classifier/exp mapping. If that classifier exists, you are basically recreating the same classifier/exp mapping instance that you retrieved, with the same values. I'm not sure I see the point...?